build_modules/lib/src/module_library.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/analyzer.dart';
+import 'package:build/build.dart';
+import 'package:meta/meta.dart';
+
+class ModuleLibrary {
```

Comment:
It would be good to have some tests for this class. There is some non trivial logic.

Reply:
None of this is new logic, it was just all private before (and therefore tested through `MetaModule.forAssets`, which it still is today).

I can add some tests though since it's now public within the package.

Reply:
I was confused by the line moved from https://github.com/dart-lang/build/pull/1649/files#diff-e6beb981c2249722d51e60e34b104f88L109

That looked like new logic to me. I'm fine with adding tests in a later PR.